test/hotspot/jtreg/compiler/loopopts/superword/TestDependencyOffsets.java

```diff
@@ -21,99 +21,12 @@
- *  - Vectors hold at least 2 elements: vector_width >= 2 * sizeofop(velt_type)
- *    -> min_vector_width = max(4, 2 * sizeofop(velt_type))
- *    -> simplifies to: vector_width >= min_vector_width
- *  - No cyclic dependency:
- *    - Access: data[i + offset] = data[i] * fac;
- *    - byte_offset = offset * sizeofop(type)
- *    - Cyclic dependency if: 0 < byte_offset < vector_width
- *
- * Note: sizeofop(type) = sizeof(type), except sizeofop(char) = 2
- *
- * Different types can lead to different vector_width. This depends on
- * the CPU-features.
- *
- * Definition:
- *     MaxVectorSize: limit through flag
- *     vector_width: limit given by specific CPU feature for a specific velt_type
- *     actual_vector_width: what is actually vectorized with
- *     min_vector_width: what is minimally required for vectorization
- *
- *     min_vector_width = max(4, 2 * sizeofop(velt_type))
- *     MaxVectorSize >= vector_width >= actual_vector_width >= min_vector_width
- *
- * In general, we cannot easily specify negative IR rules, that require no
- * vectorization to happen. We may improve the SuperWord algorithm later,
- * or some additional optimization collapses some Loads, and suddenly cyclic
- * dependency disappears, and we can vectorize.
- *
- * With '-XX:+AlignVector' we do the following:
- *
- * Must vectorize cleanly if:
- *   1) guaranteed no misalignment AND
- *   2) guaratneed no cyclic dependency
- *
- * Must not vectorize at all if:
- *   1) guaranteed misalignment AND
- *   2) guaranteed no cyclic dependency
- *
- * We could imagine a case with cyclic dependency, where C2 detects
- * that only the first load is needed, and so no vectorization is
- * required for it, and hence the store vector can be aligned.
- *
- * The alignment criteria is
- *     byte_offset % aw == 0
- * where align width (aw) is
- *     aw = min(actual_vector_width, ObjectAlignmentInBytes)
- * For simplicity, we assume that ObjectAlignmentInBytes == 8,
- * which currently can only be changed manually and then no IR
- * rule is run.
- * This allows us to do the computation statically.
- * Further, we define:
- *     aw_min = min(min_vector_width, ObjectAlignmentInBytes)
- *     aw_max = min(vector_width, ObjectAlignmentInBytes)
- *     aw_min <= aw <= aw_max
- *
- * Again, we have no cyclic dependency, except when:
- *     byte_offset > 0 and p.vector_width > byte_offset
- * Here we must ensure that:
- *     byte_offset >= MaxVectorSize
- *
- * Guaranteed no misalignment:
- *     byte_offset % aw_max == 0
- *       implies
- *         byte_offset % aw == 0
- *
- * Guaranteed misalignment:
- *     byte_offset % aw_min != 0
- *       implies
- *         byte_offset % aw != 0
- *
- */
-
 /*
  * @test id=vanilla-A
  * @bug 8298935 8308606 8310308 8312570 8310190
  * @summary Test SuperWord: vector size, offsets, dependencies, alignment.
  * @library /test/lib /
+ * @compile ../../lib/ir_framework/TestFramework.java
```

Comment:
Since this is quite a significant test change, do you wanna the this JBS number as well?